openassessment/xblock/openassessmentblock.py

```diff
@@ -1333,6 +1355,27 @@ def publish_assessment_event(self, event_name, assessment, **kwargs):
             event_data
         )
 
+    @XBlock.json_handler
+    def reset_submission(self, data, suffix=""):  # pylint: disable=unused-argument
+        """
+        Reset the student's submission.
+
+        Args:
+            data (dict): Unused parameter. Defaults to {}.
+            suffix (str, optional): Unused parameter. Defaults to ''.
+
+        Returns:
+            dict: A dictionary indication the status with keys 'success' (bool) and 'msg' (str)
+        """
+        try:
+            block_user = self.runtime.service(self, "user").get_current_user()
+            user = get_user_by_username_or_email(block_user.opt_attrs.get("edx-platform.username"))
+            reset_student_attempts(self.course_id, user, self.location, user, True)  # pylint: disable=no-member
```

Comment:
A few questions:
- does this entire block need to be inside an exception block?
- what kind of exceptions could this raise? 
- could we avoid catching broad exceptions?

Reply:
1. I would think that it is not necessary, only when calling edx-platform functions.
2. `get_user_by_username_or_email` throws `User.DoesNotExist` exception if it does not find the user. `reset_student_attempts` throws `StudentModule.DoesNotExist` exception if it does not find a submission.
3. Yes, I have already updated the code by handling specific exceptions.